### ISSUE TITLE
Unable to install distutils packages from git: "error: invalid command 'egg_info'"

### ISSUE THREAD

**Issue by synap5e** [sdispater/poetry#529]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10 64bit 
- **Poetry version**: Poetry 0.12.4, Poetry 0.12.3 / Python 3.7
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [pyproject.toml](https://gist.githubusercontent.com/synap5e/a8442dc126f0423e308e913bb293d4e9/raw/bb349645918977a4a2cbdc991d4ece230503372f/pyproject.toml)
## Issue

Attempting to install a package using `distutils` rather than `setuptools` in `setup.py` from git fails because `egg_info` is not a valid `distutils` command. 

The specific package I am trying to install is [typedload](https://github.com/ltworf/typedload), which has no dependencies and uses 
```python
from distutils.core import setup
```
in its `setup.py`.

[The full log and exception trace is here](https://gist.githubusercontent.com/synap5e/a8442dc126f0423e308e913bb293d4e9/raw/bb349645918977a4a2cbdc991d4ece230503372f/output.txt)

[The log for 0.12.3 output is here](https://gist.githubusercontent.com/synap5e/bbdfa83fc10c0ec0bdddf1305c1772eb/raw/970e29e89e956869fda18dd216337fee07baa117/output.txt)

#### Notes

I also encountered errors with installing from git regarding the directory deletion issues on windows as per #177 so my version of `poetry` has been patched to not clean up after itself 
```patch
-shutil.rmtree(tmp_dir)
+pass
```
as a temporary fix, although this does not effect this issue itself.

#### Edits
- Retested and updated issue for poetry 0.12.4

**Comment by synap5e** [comment]
Retested and edited OP for poetry 0.12.4

**Comment by nayyarv** [comment]
As a side note, you can fork and change to `from setuptools import setup` which'll allow you to continue on (and you can PR back in).

I've had to do the same, but agreed distutils needs to be supported better

**Comment by synap5e** [comment]
I checked and the typedload author isn't interested in adding setuptools as a dependency.

I don't specifically need this fixed for my use case atm as I can just grab that package from PyPy, this was just something I noticed that could cause issues in some other use cases 😄

**Comment by sdispater** [comment]
This should be fixed in the latest `0.12.6` version.
